test/integration/PluginTests.ts

```diff
@@ -0,0 +1,69 @@
+import { TASK_FLATTEN_GET_FLATTENED_SOURCE } from "@nomiclabs/buidler/builtin-tasks/task-names";
+import { BuidlerRuntimeEnvironment } from "@nomiclabs/buidler/types";
+import { assert } from "chai";
+import { ethers } from "ethers";
+// @ts-ignore
+import * as linker from "solc/linker";
+
+import ContractCompiler from "../../src/ContractCompiler";
+import deployer from "../util/ContractDeployer";
+
+declare module "mocha" {
+  interface Context {
+    env: BuidlerRuntimeEnvironment;
+  }
+}
+
+describe("Plugin integration tests", function() {
+  this.timeout(100000);
+
+  let env: BuidlerRuntimeEnvironment;
+
+  before("Buidler project setup", function() {
+    process.chdir(__dirname + "/../buidler-project");
+    process.env.BUIDLER_NETWORK = "develop";
+    env = require("@nomiclabs/buidler");
+  });
+
+  it("Test verifying deployed contract on etherscan", async () => {
+    let flattenedSource: string = await env.run(
+      TASK_FLATTEN_GET_FLATTENED_SOURCE
+    );
+
+    // need non repeating message so it makes bytecode unique
+    const message = ethers.Wallet.createRandom().address;
+    flattenedSource = flattenedSource.replace("placeholder", message);
+    const compilationResult = await new ContractCompiler(env.run).compile(
+      flattenedSource,
+      "TestContract1"
+    );
+    compilationResult.bytecode = linker.linkBytecode(
+      compilationResult.bytecode,
+      {
+        "contracts:SafeMath": "0x292FFB096f7221c0C879c21535058860CcA67f58"
+      }
+    );
+    const amount = "20";
+    const deployedAddress = await deployer.deployContract(
+      compilationResult.abi,
+      "0x" + compilationResult.bytecode,
+      amount
+    );
+    try {
+      await env.run("verify-contract", {
+        address: deployedAddress,
+        contractName: "TestContract1",
+        // libraries: JSON.stringify({
```

Comment:
Can we add a test case of sources using libraries?

Reply:
It's bugged, as it seems from their sample client. Works trough GUI though, will report it.

Reply:
Can you add a comment explaining this? And if you report it, can you link the report in the comment? Thanks!

Reply:
They are closed source, so I've reported it trough email. This is what I've sent then so you can try it trough https://ropsten.etherscan.io/sourcecode-demo.html . It fails without explanation. They said that are investigating. 
```
So original deployed source is:
https://gist.github.com/mpetrunic/5f219fd480eb633971db12ef6d17b6d0

Params are  sent as form data but i'm logging it as json are these:
https://gist.github.com/mpetrunic/322e630b332473355ad26b3dec198d86

They work when submitted trough https://ropsten.etherscan.io/verifyContract2?a=0xda628694fece2e6b79ba14cd06435da1c1a95c39 but not when sent to the api, either directly or via https://ropsten.etherscan.io/sourcecode-demo.html.
```

Reply:
Thanks for the explanation! Can you get back to us when they answer your email?

Reply:
yeah definitely!